### PR TITLE
Feature/251 import excel

### PR DESCRIPTION
**Pull request by awinia-github**

### CODE REVIEW

File: src/ATE_spyder/ate_spyder/widgets/actions_on/tests/ExcelTestWizard.py [hunk 1]
```diff
@@ -0,0 +1,650 @@
...
+Created on Mon Sep  5 18:56:05 2022
+
+@author: Zlin526F
+
+References:
+    TestWizard
+
+
+todo:  dieser Wizard stellt zur Zeit nur einen einfachen Import von einer Exel liste zur Verfügung.
+       es wird die datenbank und die zugehörigen python files für die einzelnen Test erzeugt.
```
Comment (awinia-github) [inline src/ATE_spyder/ate_spyder/widgets/actions_on/tests/ExcelTestWizard.py:12]: overall:

* remove all comments that are not necessary
* better to create github issues for the TODOs and features
* remove commented out code

File: src/ATE_spyder/ate_spyder/widgets/actions_on/tests/ExcelTestWizard.py [hunk 2]
```diff
@@ -0,0 +1,650 @@
...
+                 'Unnamed: 12': 'NaN',                     # only for test, NaN
+                 }
+
+
+SI = ['s', 'm', 'g', 'A', 'K', 'mol', 'cd', 'rad', 'sr', 'Hz', 'N', 'Pa', 'J', 'W', 'C', 'V', 'F', 'Ω', 'S', 'Wb', 'T', 'H', 'lm'
+      'lx', 'Bq', 'Gy', 'Sv', 'kat', '°C', 'Gs', '˽', '', ' ']
+
+
+@unique
+class InputColumnIndex(Enum):
```
[the remark this answers is not in the document]
Comment (Zlin526F) [reply src/ATE_spyder/ate_spyder/widgets/actions_on/tests/ExcelTestWizard.py:79]: This class isn't used in the exceltestwizard, so I'll remove it.

File: src/ATE_spyder/ate_spyder/widgets/actions_on/tests/ExcelTestWizard.py [hunk 3]
```diff
@@ -0,0 +1,650 @@
...
+    MAX = 4
+    POWER = 5
+    UNIT = 6
+    FMT = 7
+
+    def __call__(self):
+        return self.value
+
+
+@unique
```
[the remark this answers is not in the document]
Comment (Zlin526F) [reply src/ATE_spyder/ate_spyder/widgets/actions_on/tests/ExcelTestWizard.py:93]: This class isn't used in the exceltestwizard, so I'll remove it.

File: src/ATE_spyder/ate_spyder/widgets/actions_on/tests/ExcelTestWizard.py [hunk 4]
```diff
@@ -0,0 +1,650 @@
...
+    POWER = 6
+    UNIT = 7
+    MPR = 8
+    FMT = 9
+
+    def __call__(self):
+        return self.value
+
+
+class CDelegator(TestWizard.Delegator):
```
[the remark this answers is not in the document]
Comment (Zlin526F) [reply src/ATE_spyder/ate_spyder/widgets/actions_on/tests/ExcelTestWizard.py:110]: done

File: src/ATE_spyder/ate_spyder/widgets/actions_on/tests/ExcelTestWizard.py [hunk 5]
```diff
@@ -0,0 +1,650 @@
...
+            self.ForHardwareSetup.setText(test_content['hardware'])
+
+    # WithBase
+        self.WithBase.setStyleSheet("font-weight: bold")
+        self.WithBase.setText(self.project_info.active_base)
+        if test_content['base'] != '':
+            self.WithBase.setText(test_content['base'])
+
+    # Delegators
+        self.fmtDelegator = TestWizard.Delegator(valid_fmt_regex, self)                             # TODO: selectionModel isn't available!
```
[the remark this answers is not in the document]
Comment (Zlin526F) [reply src/ATE_spyder/ate_spyder/widgets/actions_on/tests/ExcelTestWizard.py:179]: done

File: src/ATE_spyder/ate_spyder/widgets/actions_on/tests/ExcelTestWizard.py [hunk 6]
```diff
@@ -0,0 +1,650 @@
...
+                    matching_items = self.table.findItems(value, QtCore.Qt.MatchExactly)[0]
+                    self._set_widget_color(matching_items, ORANGE)
+                    if addAction is not None:
+                        addAction(matching_items)
+
+        def startWithInteger(string):
+            return True if string[0].isnumeric() else False
+
+        def addMenuOverwrite(item):
+            # TODO: add ability for  overwriting
```
[the remark this answers is not in the document]
Comment (Zlin526F) [reply src/ATE_spyder/ate_spyder/widgets/actions_on/tests/ExcelTestWizard.py:383]: ok, removed

File: src/ATE_spyder/ate_spyder/widgets/actions_on/tests/ExcelTestWizard.py (hunk 1 above)
Comment (Zlin526F) [reply src/ATE_spyder/ate_spyder/widgets/actions_on/tests/ExcelTestWizard.py:12]: done

File: src/ATE_spyder/ate_spyder/widgets/actions_on/tests/ExcelTestWizard.py [hunk 7]
```diff
@@ -0,0 +1,650 @@
...
+        self.verify()
+
+
+def excel_test_dialog(project_info, selected_file):
+    newTestWizard = ExcelTestWizard(project_info, selected_file)
+    newTestWizard.exec_()
+    del(newTestWizard)
+
+
+if __name__ == "__main__":
```
[the remark this answers is not in the document]
Comment (Zlin526F) [reply src/ATE_spyder/ate_spyder/widgets/actions_on/tests/ExcelTestWizard.py]: done

**Comment by awinia-github** [comment]
will be closed with #289

**Pull request closed without merging**